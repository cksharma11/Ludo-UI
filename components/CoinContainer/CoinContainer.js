import React from 'react';
import PropTypes from 'prop-types';
import CoinContainerStyles from './CoinContainer.style';

const CoinContainer = ({ playerId, color }) => {
  return (
    <div className="coin_container" id={playerId}>
      <div className="container-border">
        <div className="container_row">
          <div id={`${color}_coin_100`} className={`coin ${color}`} />
          <div id={`${color}_coin_200`} className={`coin ${color}`} />
        </div>
        <div className="container_row">
          <div id={`${color}_coin_300`} className={`coin ${color}`} />
          <div id={`${color}_coin_400`} className={`coin ${color}`} />
        </div>
      </div>
      <style jsx>{CoinContainerStyles}</style>
    </div>
  );
};

CoinContainer.defaultProps = {
  playerId: 0,
  color: ''
};

CoinContainer.propTypes = {
  playerId: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  color: PropTypes.string
};

export default CoinContainer;
