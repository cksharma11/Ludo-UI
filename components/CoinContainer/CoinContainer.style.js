import css from 'styled-jsx/css';

const CoinContainerStyles = css`
  .coin_container {
    display: flex;
    width: 250px;
    height: 250px;
    justify-content: center;
    align-items: center;
    border: 1px solid black;
    background: #433e5d;
    /* background: #3c4844; */
  }

  .container-border {
    display: flex;
    padding: 15px;
    background: white;
  }

  .coin {
    display: flex;
    justify-content: center;
    width: 70px;
    height: 70px;
    border: 1px solid black;
    border-radius: 6px;
    margin: 2px;
  }

  .red {
    background: rgb(231, 96, 96);
  }

  .green {
    background: rgb(112, 236, 112);
  }

  .yellow {
    background: rgb(230, 230, 91);
  }

  .blue {
    background: rgb(84, 84, 235);
  }

  .dark_red {
    background: red;
  }

  .dark_green {
    background: green;
  }

  .dark_yellow {
    background: yellow;
  }

  .dark_blue {
    background: blue;
  }
`;

export default CoinContainerStyles;
