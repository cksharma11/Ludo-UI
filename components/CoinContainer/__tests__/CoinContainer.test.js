import React from 'react';
import { shallow } from 'enzyme';
import CoinContainer from '../CoinContainer';

describe('CoinContainer', () => {
  it('should render properly', () => {
    const wrapper = shallow(<CoinContainer />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
