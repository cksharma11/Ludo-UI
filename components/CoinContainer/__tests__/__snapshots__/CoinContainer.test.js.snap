// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinContainer should render properly 1`] = `
"<div id={0} className=\\"jsx-572305444 coin_container\\">
  <div className=\\"jsx-572305444 container-border\\">
    <div className=\\"jsx-572305444 container_row\\">
      <div id=\\"_coin_100\\" className=\\"jsx-572305444 coin \\" />
      <div id=\\"_coin_200\\" className=\\"jsx-572305444 coin \\" />
    </div>
    <div className=\\"jsx-572305444 container_row\\">
      <div id=\\"_coin_300\\" className=\\"jsx-572305444 coin \\" />
      <div id=\\"_coin_400\\" className=\\"jsx-572305444 coin \\" />
    </div>
  </div>
  <JSXStyle id=\\"572305444\\">
    .
    c
    o
    i
    n
    _
    c
    o
    n
    t
    a
    i
    n
    e
    r
    .
    j
    s
    x
    -
    5
    7
    2
    3
    0
    5
    4
    4
    4
    {
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    m
    s
    -
    f
    l
    e
    x
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    f
    l
    e
    x
    ;
    w
    i
    d
    t
    h
    :
    2
    5
    0
    p
    x
    ;
    h
    e
    i
    g
    h
    t
    :
    2
    5
    0
    p
    x
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    -
    p
    a
    c
    k
    :
    c
    e
    n
    t
    e
    r
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    j
    u
    s
    t
    i
    f
    y
    -
    c
    o
    n
    t
    e
    n
    t
    :
    c
    e
    n
    t
    e
    r
    ;
    -
    m
    s
    -
    f
    l
    e
    x
    -
    p
    a
    c
    k
    :
    c
    e
    n
    t
    e
    r
    ;
    j
    u
    s
    t
    i
    f
    y
    -
    c
    o
    n
    t
    e
    n
    t
    :
    c
    e
    n
    t
    e
    r
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    a
    l
    i
    g
    n
    -
    i
    t
    e
    m
    s
    :
    c
    e
    n
    t
    e
    r
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    -
    a
    l
    i
    g
    n
    :
    c
    e
    n
    t
    e
    r
    ;
    -
    m
    s
    -
    f
    l
    e
    x
    -
    a
    l
    i
    g
    n
    :
    c
    e
    n
    t
    e
    r
    ;
    a
    l
    i
    g
    n
    -
    i
    t
    e
    m
    s
    :
    c
    e
    n
    t
    e
    r
    ;
    b
    o
    r
    d
    e
    r
    :
    1
    p
    x
     
    s
    o
    l
    i
    d
     
    b
    l
    a
    c
    k
    ;
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    #
    4
    3
    3
    e
    5
    d
    ;
    }
    .
    c
    o
    n
    t
    a
    i
    n
    e
    r
    -
    b
    o
    r
    d
    e
    r
    .
    j
    s
    x
    -
    5
    7
    2
    3
    0
    5
    4
    4
    4
    {
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    m
    s
    -
    f
    l
    e
    x
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    f
    l
    e
    x
    ;
    p
    a
    d
    d
    i
    n
    g
    :
    1
    5
    p
    x
    ;
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    w
    h
    i
    t
    e
    ;
    }
    .
    c
    o
    i
    n
    .
    j
    s
    x
    -
    5
    7
    2
    3
    0
    5
    4
    4
    4
    {
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    m
    s
    -
    f
    l
    e
    x
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    f
    l
    e
    x
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    -
    p
    a
    c
    k
    :
    c
    e
    n
    t
    e
    r
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    j
    u
    s
    t
    i
    f
    y
    -
    c
    o
    n
    t
    e
    n
    t
    :
    c
    e
    n
    t
    e
    r
    ;
    -
    m
    s
    -
    f
    l
    e
    x
    -
    p
    a
    c
    k
    :
    c
    e
    n
    t
    e
    r
    ;
    j
    u
    s
    t
    i
    f
    y
    -
    c
    o
    n
    t
    e
    n
    t
    :
    c
    e
    n
    t
    e
    r
    ;
    w
    i
    d
    t
    h
    :
    7
    0
    p
    x
    ;
    h
    e
    i
    g
    h
    t
    :
    7
    0
    p
    x
    ;
    b
    o
    r
    d
    e
    r
    :
    1
    p
    x
     
    s
    o
    l
    i
    d
     
    b
    l
    a
    c
    k
    ;
    b
    o
    r
    d
    e
    r
    -
    r
    a
    d
    i
    u
    s
    :
    6
    p
    x
    ;
    m
    a
    r
    g
    i
    n
    :
    2
    p
    x
    ;
    }
    .
    r
    e
    d
    .
    j
    s
    x
    -
    5
    7
    2
    3
    0
    5
    4
    4
    4
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    r
    g
    b
    (
    2
    3
    1
    ,
    9
    6
    ,
    9
    6
    )
    ;
    }
    .
    g
    r
    e
    e
    n
    .
    j
    s
    x
    -
    5
    7
    2
    3
    0
    5
    4
    4
    4
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    r
    g
    b
    (
    1
    1
    2
    ,
    2
    3
    6
    ,
    1
    1
    2
    )
    ;
    }
    .
    y
    e
    l
    l
    o
    w
    .
    j
    s
    x
    -
    5
    7
    2
    3
    0
    5
    4
    4
    4
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    r
    g
    b
    (
    2
    3
    0
    ,
    2
    3
    0
    ,
    9
    1
    )
    ;
    }
    .
    b
    l
    u
    e
    .
    j
    s
    x
    -
    5
    7
    2
    3
    0
    5
    4
    4
    4
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    r
    g
    b
    (
    8
    4
    ,
    8
    4
    ,
    2
    3
    5
    )
    ;
    }
    .
    d
    a
    r
    k
    _
    r
    e
    d
    .
    j
    s
    x
    -
    5
    7
    2
    3
    0
    5
    4
    4
    4
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    r
    e
    d
    ;
    }
    .
    d
    a
    r
    k
    _
    g
    r
    e
    e
    n
    .
    j
    s
    x
    -
    5
    7
    2
    3
    0
    5
    4
    4
    4
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    g
    r
    e
    e
    n
    ;
    }
    .
    d
    a
    r
    k
    _
    y
    e
    l
    l
    o
    w
    .
    j
    s
    x
    -
    5
    7
    2
    3
    0
    5
    4
    4
    4
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    y
    e
    l
    l
    o
    w
    ;
    }
    .
    d
    a
    r
    k
    _
    b
    l
    u
    e
    .
    j
    s
    x
    -
    5
    7
    2
    3
    0
    5
    4
    4
    4
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    b
    l
    u
    e
    ;
    }
  </JSXStyle>
</div>"
`;
